tour_model_eval/user_input_request_times_all_users.ipynb

```diff
@@ -0,0 +1,493 @@
+    "        req_day.append(NaN)\n",
+    "        req_month.append(NaN) \n",
+    "        continue\n",
+    "    sim.bin_data()\n",
+    "    sim.delete_bins()\n",
+    "    bins = sim.bins\n",
+    "    \n",
+    "    # collect requested trips and common trips(no need to request) indices above cutoff\n",
+    "    ab_trip_ls = []\n",
+    "    no_req_trip_ls = []\n",
+    "    for bin in bins:\n",
+    "        early_trip_index, index = evaluation.find_first_trip(filter_trips,bin)\n",
+    "        ab_trip_ls.append(early_trip_index)\n",
+    "        \n",
+    "        for k in range(len(bin)):\n",
+    "            if k != index:\n",
+    "                no_req_trip_idx = bin[k]\n",
+    "                no_req_trip_ls.append(no_req_trip_idx)\n",
+    "\n",
+    "\n",
+    "    \n",
+    "    # bins below cutoff\n",
+    "    bl_bins = sim.below_cutoff\n",
+    "    \n",
+    "    # collect requested trips indices below cutoff\n",
+    "    bl_trip_ls = []\n",
+    "    for bin in bl_bins:\n",
+    "        for trip_index in bin:\n",
+    "            bl_trip_ls.append(trip_index)\n",
+    "            \n",
+    "    # whole list of requested trips indices\n",
+    "    req_trips_ls=ab_trip_ls+bl_trip_ls\n",
+    "    \n",
+    "    \n",
+    "    # collect request times in a day\n",
+    "    bin_day = evaluation.bin_date(req_trips_ls,filter_trips,day=True)\n",
+    "    req_day_ls = []\n",
+    "    for bin in bin_day:\n",
+    "        req_day_ls.append(len(bin))\n",
+    "               \n",
+    "    # collect 0 request days \n",
+    "    for trip_index in no_req_trip_ls:\n",
+    "        trip = filter_trips[trip_index]\n",
+    "        match = False\n",
+    "        for bin in bin_day:\n",
+    "            if evaluation.match_day(trip,bin,filter_trips):\n",
+    "                match = True\n",
+    "                break\n",
+    "        if not match:\n",
+    "            req_day_ls.append(0)\n",
+    "            \n",
+    "    # collect request times in a day for every user\n",
+    "    req_day.append(req_day_ls)\n",
+    "    \n",
+    "    # collect user input request proportion in a day\n",
+    "    filter_trips_df = pd.DataFrame(filter_trips)\n",
+    "    filter_trips_idx_ls = filter_trips_df.index.values.tolist()\n",
+    "    bin_filter_trips_day = evaluation.bin_date(filter_trips_idx_ls,filter_trips,day=True)\n",
+    "    propor_single_user = []\n",
+    "    for valid_trips_bin in bin_filter_trips_day:\n",
+    "        match = False\n",
+    "        for req_trips_bin in bin_day:\n",
+    "            req_trip = filter_trips[req_trips_bin[0]]\n",
+    "            if evaluation.match_day(req_trip,valid_trips_bin,filter_trips):\n",
+    "                proportion = round(len(req_trips_bin)/len(valid_trips_bin), 2)\n",
+    "                propor_single_user.append(proportion)\n",
+    "                match = True\n",
+    "                break\n",
```

Comment:
This is the problem with the current `match_day` implementation. Reading this appears that you compare one trip to a list to find the matches, and then you determine the proportion for that user? Where is the iteration across days?

This looks wrong, but it has no comments for explanation.